internal/bundleutil/bundle.go

```diff
@@ -0,0 +1,39 @@
+package bundleutil
+
+import (
+	"encoding/json"
+	"fmt"
+
+	bsemver "github.com/blang/semver/v4"
+
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	"github.com/operator-framework/operator-registry/alpha/property"
+
+	ocv1alpha1 "github.com/operator-framework/operator-controller/api/v1alpha1"
+)
+
+func GetVersion(b declcfg.Bundle) (*bsemver.Version, error) {
+	for _, p := range b.Properties {
+		if p.Type == property.TypePackage {
+			var pkg property.Package
+			if err := json.Unmarshal(p.Value, &pkg); err != nil {
+				return nil, fmt.Errorf("error unmarshalling package property: %w", err)
+			}
+			vers, err := bsemver.Parse(pkg.Version)
+			if err != nil {
+				return nil, err
+			}
+			return &vers, nil
+		}
+	}
+	return nil, fmt.Errorf("no package property found in bundle %q", b.Name)
+}
+
+// MetadataFor returns a BundleMetadata for the given bundle. If the provided bundle is nil,
+// this function panics. It is up to the caller to ensure that the bundle is non-nil.
+func MetadataFor(bundleName string, bundleVersion *bsemver.Version) *ocv1alpha1.BundleMetadata {
```

Comment:
would it make sense to wrap the bundle in an internal struct with this added method? I can see both ways working, I just don't know how "first class" this method needs to be.

Reply:
`catalogmetadata.Bundle` was the wrapper that this PR removed, somewhat purposefully. What I had noticed is that we started losing parity with the FBC data model (we put channel membership and deprecation information in that wrapper). And that seemed like a recipe for more spaghetti-making when FBC evolves new types and semantics.

So I opted for functions over `declcfg.Bundle` rather than a wrapper type with methods.